### PR TITLE
chore: add proposal issue template

### PR DESCRIPTION
**Pull request by devigned**
**What this PR does / why we need it**:

This PR introduces a proposals template. We need it because the repo does not have one. By placing a template, we can encourage new proposal and provide a standard format for such proposals.

**Release note**:

```release-note
add proposal issue template based on CAPI template
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
/approve
/lgtm

once the tests are fixed in a different PR, this would be ready to get merged with a retest

**Comment by vincepri** [comment]
What are we trying to accomplish here? You might want to look at the issue template in CAPI, for a lighter process on proposals https://github.com/kubernetes-sigs/cluster-api/issues/new?template=proposal.md

**Comment by devigned** [comment]
So the goal here is to provide a template for proposing changes which are scoped to this project.

Perhaps, like CAPI, this PR should change the issue templates to include both the lightweight proposal and the heavy weight version.

Thoughts?

**Comment by CecileRobertMichon** [comment]
@justaugustus @vincepri for context, @devigned had a design proposal he wants to share and asked me what format to use and I suggested following the CAEP format and we thought it'd be a good idea to put the template in the repo. Issues are great for lightweight ideas but sometimes you need a bit more detail if the change is significant in scope. How about a process like this: 1) open a github issue with the idea 2) get feedback on if there's a need for it 3) open a CAEP PR if the scope is big and maintainers request it, wdyt?

**Comment by devigned** [comment]
Based on the feedback today in the CAPZ office hours, I've change this to only be an issue template for lightweight proposals and a ptr to CAEP for a more centralized process for larger proposals.

File: .github/ISSUE_TEMPLATE/proposal.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+name: RFE Proposal
+about: Propose larger efforts, breaking changes, or new features
+
+---
+
+> ⚠️ Cluster API maintainers can ask to turn an issue-proposal into a [CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/YYYYMMDD-template.md) when necessary. This is to be expected for large changes that impact multiple components, breaking changes, or new large features.
```
Comment (CecileRobertMichon) [inline .github/ISSUE_TEMPLATE/proposal.md]:
```suggestion
> ⚠️ Cluster API Azure maintainers can ask to turn an issue-proposal into a [CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/YYYYMMDD-template.md) when necessary. This is to be expected for large changes that impact multiple components, breaking changes, or new large features.
```

File: .github/ISSUE_TEMPLATE/proposal.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
+name: RFE Proposal
```
Comment (CecileRobertMichon) [inline .github/ISSUE_TEMPLATE/proposal.md]: what's RFE?

Comment (devigned) [reply .github/ISSUE_TEMPLATE/proposal.md]: Idk... it was in the CAPI template

Comment (CecileRobertMichon) [reply .github/ISSUE_TEMPLATE/proposal.md]: can we rename it to something that's more meaningful to us? Maybe `Enhancement Proposal`?

**Comment by CecileRobertMichon** [comment]
@devigned don't forget to squash commits, this repo doesn't squash on merge...

File: .github/ISSUE_TEMPLATE/proposal.md (hunk 2 above)
Comment (vincepri) [reply .github/ISSUE_TEMPLATE/proposal.md]: Request for enhancement

Comment (devigned) [reply .github/ISSUE_TEMPLATE/proposal.md]: RFE, spell out RFE, or `Enhancement Proposal`? Ready... vote!

**Comment by devigned** [comment]
@CecileRobertMichon, thx for the reminder. Will do when we nail down the issue template name.

**Comment by devigned** [comment]
I'm about to hop on a long flight with possible losses of connectivity, so I've rebased and left the template as RFE.

File: .github/ISSUE_TEMPLATE/proposal.md (hunk 2 above)
Comment (CecileRobertMichon) [reply .github/ISSUE_TEMPLATE/proposal.md]: I vote `Enhancement Proposal`, I don't think acronyms are friendly to newcomers. And if I we both didn't know what it was, chances are some contributors won't know either.

Comment (devigned) [reply .github/ISSUE_TEMPLATE/proposal.md]: Anyone else have a preference?

Comment (cpanato) [reply .github/ISSUE_TEMPLATE/proposal.md]: `Enhancement Proposal`

Comment (devigned) [reply .github/ISSUE_TEMPLATE/proposal.md]: I've changed the issue template to be titled `Enhancement Proposal`.

**Pull request merged**